src/lib/Menu/ContentRightSidebarBuilder.php

```diff
@@ -388,19 +406,70 @@ private function addRevealMenuItem(ItemInterface $menu): void
     /**
      * @param \Knp\Menu\ItemInterface $menu
      */
-    private function addHideMenuItem(ItemInterface $menu): void
+    private function addHideMenuItem(ItemInterface $menu, bool $canHide): void
     {
+        $attributes = [
+            'data-actions' => 'hide',
+            'class' => 'ez-btn--hide',
+        ];
+
         $menu->addChild(
             $this->createMenuItem(
                 self::ITEM__HIDE,
                 [
                     'extras' => ['icon' => 'hide'],
-                    'attributes' => [
-                        'data-actions' => 'hide',
-                        'class' => 'ez-btn--hide',
-                    ],
+                    'attributes' => $canHide
+                        ? $attributes
+                        : array_merge($attributes, ['disabled' => 'disabled']),
                 ]
             )
         );
     }
+
+    private function canCreateAnywhere(Content $content): bool
+    {
+        $createPolicies = $this->permissionResolver->hasAccess(
+            'content',
+            'create'
+        );
+
+        if (is_bool($createPolicies)) {
+            return $createPolicies;
+        }
+
+        foreach ($createPolicies as $createPolicy) {
+            $sectionId = $content->contentInfo->sectionId;
+            if (
+                !empty($createPolicy['limitation']) &&
+                $createPolicy['limitation'] instanceof SectionLimitation &&
+                !in_array($sectionId, $createPolicy['limitation']->limitationValues)
+            ) {
+                return false;
+            }
+
+            if (empty($createPolicy['policies'])) {
+                return true;
+            }
+
+            foreach ($createPolicy['policies'] as $policy) {
+                foreach ($policy->limitations as $limitation) {
+                    if ($limitation instanceof SectionLimitation && !in_array($sectionId, $limitation->limitationValues)) {
+                        return false;
+                    }
+
+                    if ($limitation instanceof ContentType && !in_array($content->getContentType()->id, $limitation->limitationValues)) {
+                        return false;
+                    }
+
+                    $commonArray = \array_intersect($content->getVersionInfo()->languageCodes, $limitation->limitationValues);
+
+                    if ($limitation instanceof LanguageLimitation && \count($commonArray) === 0) {
+                        return false;
+                    }
+                }
+            }
+        }
```

Comment:
I believe we can't make those `in_array` checks strict (third parameter)?

Reply:
We get data from the database and don't have any cast that I think we can't check strict type